browser/android/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 3.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/android/brave_cosmetic_resources_tab_helper.h"
+
+#include <string>
+
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/ad_block_regional_service_manager.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/browser/web_contents/web_contents_impl.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/renderer/render_frame.h"
+
+
+namespace {
+bool IsAdBlockEnabled(content::WebContents* contents, const GURL& url) {
+  Profile* profile = static_cast<Profile*>(contents->GetBrowserContext());
+
+  auto* map = HostContentSettingsMapFactory::GetForProfile(profile);
+  brave_shields::ControlType control_type =
+      brave_shields::GetAdControlType(map, contents->GetLastCommittedURL());
+  if (brave_shields::GetBraveShieldsEnabled(map, url) &&
+      control_type != brave_shields::ALLOW) {
+    content::RenderFrameHost::AllowInjectingJavaScript();
+
+    return true;
+  }
+
+  return false;
+}
```

Comment:
Does that check return false if the general shields are down?

Reply:
Yep! It's just this:

```
bool ShouldDoCosmeticFiltering(HostContentSettingsMap* map, const GURL& url) {
  return base::FeatureList::IsEnabled(features::kBraveAdblockCosmeticFiltering)
      && GetBraveShieldsEnabled(map, url)
      && (GetCosmeticFilteringControlType(map, url) != ControlType::ALLOW);
}
```

So you'll probably also want the additional `content::RenderFrameHost::AllowInjectingJavaScript()` check.

Reply:
made the `ShouldDoCosmeticFiltering` check instead of what I had and moved `content::RenderFrameHost::AllowInjectingJavaScript()` to the place when we actually need to inject a script, because before I basically allowed for any website when adblock is enabled.